Handlers/Message.py

```diff
@@ -108,3 +104,17 @@ def get_trending_hashtags(self):
             trending_hashtags.append({'hashtag': hashtag['hashtag'], 'position': i + 1})
         return jsonify(trending_hashtags)
 
+    def _vote_message(self, mid, upvote):
+        user = get_jwt_identity()
+        uid = UserDAO().get_user_by_username(user)['uid']
+        try:
+            self.dao.vote_message(mid, uid, upvote)
+            msg = 'Successfully added vote'
+        except:
+            self.dao.conn.rollback()
+            deleted = self.dao.remove_vote(mid, uid, upvote)
```

Comment:
If there is a rollback how come you have to delete the vote? Doesn't rollback handle this?

Reply:
Esto corre si ya existe un voto, en ese caso hay un databaseerror and the transaction is aborted. Tienes que hacer un rollback para volver la base de dato a un estado stable por si quieres hacer otra operacion